### ISSUE TITLE
Can't find duplicacy preferences file

### ISSUE THREAD

**Issue by alexbaucom17** [christophetd/duplicacy-autobackup#3]
I am trying to set up this container and getting the following error in the docker logs: "Failed to read the preference file from repository /data: open /data/.duplicacy/preferences: no such file or directory". It looks like this is preventing the backup from running.

**Comment by alexbaucom17** [comment]
Briefly poking around the duplicay wiki for info about the preferences file shows that it is a file that gets created upon initialization. I am pretty new to docker and duplicacy but it looks like maybe there should be a `duplicacy init` somewhere in the DockerFile?

EDIT: Jk looks like the init happens in the backup.sh script.

**Comment by christophetd** [comment]
Thanks a lot for the report. Do you happen to have an error prior to that in the logs? Something like 'invalid AWS credentials'. If the `duplicacy init` command fails, the preference file wouldn't created hence leading to the error you're describing.

We need better error handling for that, I've opened #4 to track it.

**Comment by alexbaucom17** [comment]
The logs did not provide too much info for me. That error was the only thing the logs gave me initially, but I actually was able to get it working with a bit of tinkering.

The command I used to start my container was 
`docker run -d --name duplicacy-test \`
`-v /media/raid/testfolder:/data \`
`-e BACKUP_NAME='test-backup' \`
`-e BACKUP_LOCATION='b2://MyBackupTest' \`
`-e BACKUP_SCHEDULE='0 2 * * *' \`
`-e DUPLICACY_BACKUP_OPTIONS='-threads 4 -stats -limit-rate 200' \`
`-e BACKUP_ENCRYPTION_KEY='mytestkey' \`
`-e B2_ID='#########' \`
`-e B2_KEY='######################' \`
`christophetd/duplicacy-autobackup`

If /media/raid/test was uninitialized, all I would get was the failed to read preference file if I just ran the container as is. 

Eventually, I tried entering the container in interactive mode and ran backup.sh directly from in the container, and this produced a more meaningful output. The first time I ran it, I got an error that my encryption key was too short because I initially had it as something like "test" and 8 characters is required.

Once I changed the encryption key, the backup initialized correctly and began running. I have since removed and restarted the container a number of different times and it has worked properly each time now that the preferences file is initialized correctly in that directory. So I am not sure if the password was the problem and that just didn't get logged initially, or if there was something weird for me with doing the initialization automatically in the container. I also had some permissions issues going on with my raid array that I think I have now fixed, so it is possible that caused the problem too.

This afternoon I am going to try setting up a new container on a different uninitialized folder. I'll let you know how that goes.

**Comment by alexbaucom17** [comment]
Okay so I did some more testing and it looks like this was definitely an issue on my end - either the password being too short or my permissions being messed up, and the logs just didn't give me much info so I was confused. I started up a new container pointing at a different test folder backing up to the same B2 bucket that I had already set up and that initialized correctly and worked just fine. Then I also tried a new container on another folder backing up to a new B2 bucket and that also worked correctly.
